docs/wikibook.md

```diff
@@ -61,3 +62,244 @@ void playGame(ArrayList<Character> characters, ArrayList<Item> items, ArrayList<
+   }
+   Console.readLine();
+}
+```
+
+### Q5 - Create 'teleport (location)' command.
+
+There are multiple ways this may go. They could be nice and just ask you to input room ID's, which is nice and easy, and would be worth a couple marks. I expect that they would give you the room ID's, and then have you ask the user where they would like to go and do it that way, which makes things slightly more tricky but still not bad. They will probably ask for a new function to be made as well. I'm guessing it would be a middle to end question worth a fair amount of marks. I have coded the "tricker" version.
+
+```java
+void teleportToRoom(ArrayList<Character> characters, ArrayList<Place> places, ArrayList<Item> items) {
+   int response;
+
+   Console.writeLine();
+   Console.writeLine("1 = Blue room");
+   Console.writeLine("2 = Store cupboard");
+   Console.writeLine("3 = Bare bed room");
+   Console.writeLine("4 = Empty corridor");
+   Console.writeLine("5 = Guardroom");
+   Console.writeLine("6 = Flag Jail cell");
+   Console.writeLine("7 = Skeleton Jail cell");
+   Console.writeLine("8 = Cellar");
+   Console.writeLine("Here are a list of rooms. Enter the number relating to the room you wish to teleport to: ");
+
+   //If an invalid response, exit the function, otherwise teleport the player
+   try {
+      response = Integer.parseInt(Console.readLine().trim());
+      characters.get(0).currentLocation = response;
+      say("You have been teleported.");
+      
+      //Describe the room and its items
+      Console.writeLine(places.get(characters.get(0).currentLocation - 1).description);
+      displayGettableItemsInLocation(items, characters.get(0).currentLocation);
+   } catch (Exception e) {
+      say("Not a valid room to teleport to.");
+      return;
+   }
+}
+```
+
+```diff
+void playGame(ArrayList<Character> characters, ArrayList<Item> items, ArrayList<Place> places) {
+  [']
+  [']
+  [']
++        case "teleport":
++           teleportToRoom(characters, places, items);
++           break;
+            [']
+            [']
+            [']
+         default:
+            Console.writeLine("Sorry, you don't know how to " + command + ".");
+      }
+   }
+   Console.readLine();
+}
+```
+
+### Q6 - Create 'save (filename)' command.
+
+It's a tedious rewrite of the loadGame function, but with some changes so you write to a file instead of reading from a file (as expected duh). If it was to come up it would be a ending question as it is a long one, but I have a feeling it won't be in the exam.
+
+```java
+boolean saveGame(String filename, ArrayList<Character> characters, ArrayList<Item> items, ArrayList<Place> places) {
+   int count;
+   try {
+      FileOutputStream binaryWriter = new FileOutputStream(filename);
+      DataOutputStream writer = new DataOutputStream(binaryWriter);
+      writer.writeInt(characters.size());
+      
+      for (count = 0; count < characters.size() ; count++) {
+      	writer.writeInt(characters.get(count).id);
```

Comment:
Inconsistent indentation - I'll fix this once I've merged it.

Reply:
I do love Eclipse sometimes.